### PR TITLE
Add `mcmod.info` to gettingstarted/structuring.md

### PR DESCRIPTION
**Pull request by howtonotwin**
Define what it is, what it does, and what it can define.

### CODE REVIEW

File: docs/gettingstarted/structuring.md [hunk 1]
```diff
@@ -14,6 +14,50 @@ Pick a unique package name. If you own a URL associated with your project, you c
 
 After the top level package (if you have one) you append a unique name for your mod, such as `examplemod`. In our case it will end up as `com.example.examplemod`.
 
+The `mcmod.info` file
+-------------------
+
+This file defines the metadata of your mod. This file lists information about your mod that users may view from the main screen of the game through the Mods button. A single info file can describe several modids. If the a mod has set `Mod.useMetadata = true`, the `mcmod.info` will override the corresponding metadata in the `@Mod` annotation. You should define every property (except `updateUrl`) so there is never an error when parsing.
```
Comment (williewillus) [inline docs/gettingstarted/structuring.md]: > Mod.useMetadata

where's that, in @Mod? clarify that

> You should define every property

i don't think this is necessary, just note in your table below what the minimal necessities are

File: docs/gettingstarted/structuring.md [hunk 2]
```diff
@@ -14,6 +14,50 @@ Pick a unique package name. If you own a URL associated with your project, you c
...
+|  screenshots | [string] |   `[]`   | A list of images to be shown on the info page. Currently unimplemented. |
+|       parent |  string  |   `""`   | The modid of a parent mod, if applicable. Using this allows modules of another mod to be listed under it in the info page, like BuildCraft.
+| requiredMods | [string] |   `[]`   | A list of modids. If one is missing, the game will crash. This **does not affect the _ordering_ of mod loading!** To specify ordering as well as requirement, have a coupled entry in `dependencies`. |
+| dependencies | [string] |   `[]`   | A list of modids. All of the listed mods will load *before* this one. If it's not present, nothing happens. |
+|   dependants | [string] |   `[]`   | A list of modids. All of the listed mods will load *after* this one. If it's not present, nothing happens. |
+| useDependencyInormation |  boolean |  `false` | If true and (of course) `Mod.useMetadata == true`, the above 3 lists of dependencies will be used. If not, the above 3 do nothing. |
+
+A good example `mcmod.info` that uses many of these properties is [BuildCraft](http://github.com/BuildCraft/BuildCraft/blob/8.0.x/buildcraft_resources/mcmod.info).
```
Comment (williewillus) [inline docs/gettingstarted/structuring.md]: reupload it as an anonymous gist incase the bc repo shifts or the branch gets changed in the future

File: docs/gettingstarted/structuring.md [hunk 3]
```diff
@@ -14,6 +14,50 @@ Pick a unique package name. If you own a URL associated with your project, you c
...
+|  screenshots | [string] |   `[]`   | A list of images to be shown on the info page. Currently unimplemented. |
+|       parent |  string  |   `""`   | The modid of a parent mod, if applicable. Using this allows modules of another mod to be listed under it in the info page, like BuildCraft.
+| requiredMods | [string] |   `[]`   | A list of modids. If one is missing, the game will crash. This **does not affect the _ordering_ of mod loading!** To specify ordering as well as requirement, have a coupled entry in `dependencies`. |
+| dependencies | [string] |   `[]`   | A list of modids. All of the listed mods will load *before* this one. If it's not present, nothing happens. |
+|   dependants | [string] |   `[]`   | A list of modids. All of the listed mods will load *after* this one. If it's not present, nothing happens. |
+| useDependencyInormation |  boolean |  `false` | If true and (of course) `Mod.useMetadata == true`, the above 3 lists of dependencies will be used. If not, the above 3 do nothing. |
```
Comment (williewillus) [inline docs/gettingstarted/structuring.md]: take out the (of course). it's unnecessary fluff

Comment (williewillus) [inline docs/gettingstarted/structuring.md]: I would also move this to be above the prior 3

File: docs/gettingstarted/structuring.md (hunk 1 above)
Comment (howtonotwin) [inline docs/gettingstarted/structuring.md]: Thanks for the help.

File: docs/gettingstarted/structuring.md [hunk 4]
```diff
@@ -14,6 +14,50 @@ Pick a unique package name. If you own a URL associated with your project, you c
...
+      "version": "1.0.0.0",
+      "mcversion": "1.10.2",
+      "logoFile": "assets/examplemod/textures/logo.png",
+      "url": "minecraftforge.net/",
+      "updateJSON": "minecraftforge.net/versions.json",
+      "authorlist": ["Author"],
+      "credits": "I'd like to thank my mother and father.",
+    }]
+
+The default Gradle configuration replaces `${version}` with the project version, and `${mcversion}` with the Minecraft version, but *only* within `mcmod.info`, so you should use those instead of directly writing them out. Here is a table of attributes that may be given to a mod; a default of `required` means there is no default and the absence of the property causes problems:
```
Comment (williewillus) [inline docs/gettingstarted/structuring.md]: s/problems/an error

**Pull request merged**